py/desisim/targets.py

```diff
@@ -128,29 +128,39 @@ def sample_objtype(nobj, flavor):
 
 #- multiprocessing needs one arg, not multiple args
 def _wrap_get_targets(args):
-    nspec, flavor, tileid = args
-    return get_targets(nspec, flavor, tileid)
+    nspec, flavor, tileid, randseed = args
+    return get_targets(nspec, flavor, tileid, randseed=randseed)
     
-def get_targets_parallel(nspec, flavor, tileid=None, nproc=None):
+def get_targets_parallel(nspec, flavor, tileid=None, nproc=None, randseed=None):
     import multiprocessing as mp
     if nproc is None:
         nproc = mp.cpu_count() // 2
-    
+
     #- don't bother with parallelism if there aren't many targets
     if nspec < 20:
         log.debug('Not Parallelizing get_targets for only {} targets'.format(nspec))
+        np.random.seed(randseed)
         return get_targets(nspec, flavor, tileid)
```

Comment:
I think you still need to pass randseed to get_targets() here.  Then the naked call to np.random.seed(randseed) (previous line) can probably go away.  There should also be a test covering the case of nspec<20.